packages/wdio-browserstack-service/src/launcher.ts

```diff
@@ -58,7 +63,47 @@ export default class BrowserstackLauncherService implements Services.ServiceInst
         }
     }
 
-    onPrepare (config?: Options.Testrunner, capabilities?: Capabilities.RemoteCapabilities) {
+    async onPrepare (config?: Options.Testrunner, capabilities?: Capabilities.RemoteCapabilities) {
+        /**
+         * Upload app to BrowserStack if valid file path to app is given.
+         * Update app value of capability directly if app_url, custom_id, shareable_id is given
+         */
+        if (!this._options.app) {
+            log.info('app is not defined in browserstack-service config, skipping ...')
+        } else {
+            let app: App = {}
+            let appConfig: AppConfig | string = this._options.app
+
+            try {
+                app = await this._validateApp(appConfig)
+            } catch (error: unknown){
+                log.error(error)
+                return process.emit('SIGTERM')
```

Comment:
We should throw a `SevereServiceError` here which causes the service to stop the test run. You can import it via:

```js
import { SevereServiceError } from 'webdriverio'
```

Add `webdriverio` as peer dependency to make this import possible.